sqlglot/dialects/duckdb.py

```diff
@@ -96,6 +97,21 @@ def _json_format_sql(self: generator.Generator, expression: exp.JSONFormat) -> s
     return f"CAST({sql} AS TEXT)"
 
 
+def _ensure_utf8(charset: exp.Literal) -> None:
+    if charset.name.lower() != "utf-8":
+        raise UnsupportedError(f"Unsupported charset {charset}")
+
+
+def _decode_sql(self: generator.Generator, expression: exp.Decode) -> str:
+    _ensure_utf8(expression.args["charset"])
+    return self.func("DECODE", expression.this, expression.args.get("replace"))
+
+
+def _encode_sql(self: generator.Generator, expression: exp.Encode) -> str:
+    _ensure_utf8(expression.args["charset"])
+    return f"ENCODE({self.sql(expression, 'this')})"
+
```

Comment:
This section is (almost) copy-pasted from presto.py -- can we move these in dialect.py and reuse them in both presto and duckdb instead of duplicating it here?